src/main/java/in/org/projecteka/hiu/dataflow/LocalDataStore.java

```diff
@@ -13,48 +13,52 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
+import java.util.Optional;
 
 public class LocalDataStore {
+    private static final Logger logger = Logger.getLogger(LocalDataStore.class);
 
     public Mono<Void> serializeDataToFile(DataNotificationRequest dataNotificationRequest, Path outFileName) {
-        return Mono.create(monoSink -> {
-            byte[] bytes = contentFromRequest(dataNotificationRequest);
-            ByteBuffer byteBuffer = ByteBuffer.wrap(bytes);
-            //TODO: find location from application properties. also create the filename under a directory thats relevant to transaction
+        return Mono.create(monoSink ->
+                contentFromRequest(dataNotificationRequest)
+                        .ifPresent(bytes -> {
```

Comment:
What happens if Optional is not present? Should throw error.

Reply:
Handled now, before my change it was handled, that's why I left it as it was.